NetSDK/Services/Impressions/Classes/InMemoryTreatmentLog.cs

```diff
@@ -0,0 +1,77 @@
+﻿using log4net;
+using Splitio.Domain;
+using Splitio.Services.Impressions.Interfaces;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Splitio.Services.Impressions.Classes
+{
+    public class InMemoryTreatmentLog : ITreatmentLog
+    {
+        private ConcurrentDictionary<String, ConcurrentQueue<KeyImpression>> cache;
+        private int maximumNumberOfKeysToCachePerTest;
+        private object lockObject = new Object();
+        protected static readonly ILog Logger = LogManager.GetLogger(typeof(InMemoryTreatmentLog));
+
+        public InMemoryTreatmentLog(ConcurrentDictionary<string, ConcurrentQueue<KeyImpression>> cache = null, int maximumNumberOfKeysToCachePerTest = -1)
+        {
+            this.maximumNumberOfKeysToCachePerTest = maximumNumberOfKeysToCachePerTest;
+            this.cache = cache ?? new ConcurrentDictionary<string, ConcurrentQueue<KeyImpression>>();
+        }
+
+        protected virtual void NotifyEviction(string feature, ConcurrentQueue<KeyImpression> impressions) { }
+
+
+        public ConcurrentDictionary<String, ConcurrentQueue<KeyImpression>> FetchAllAndClear()
+        {
+            lock (lockObject)
+            {
+                var existingImpressions = new ConcurrentDictionary<String, ConcurrentQueue<KeyImpression>>(cache);
+                cache = new ConcurrentDictionary<String, ConcurrentQueue<KeyImpression>>();
+                return existingImpressions;
+            }
+        }
+
+        public void Log(string id, string feature, string treatment, long time)
+        {
+            if (id == null || feature == null || treatment == null || time < 0)
+            {
+                return;
+            }
+
+            lock (lockObject)
```

Comment:
@patricioe Discussed with @jwaimann the `ConcurrentQueue` seems like is not the same implementation than `BlockingQueue` in java (you can not set the size), we think that we would need to create a wrapper class for protect writings overrides and copies as well, current implementation tries to do that what we think that we should encapsulate that functionality if the `ConcurrentQueue` doesn't provide that